src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -1983,12 +1985,15 @@ private void VisitSelect(Expression operand)
             }
         }
 
-        private string RemoveTransperentIdentifiersIfNeeded(MemberExpression memberExpression)
+        private string RemoveTransparentIdentifiersIfNeeded(MemberExpression memberExpression)
         {
             var selectPath = GetSelectPath(memberExpression);
-            while (selectPath.StartsWith("<>h__TransparentIdentifier"))
+            while (selectPath.StartsWith(TransparentIdentifier))
             {
-                selectPath = selectPath.Substring(28);
+                selectPath = selectPath.Substring(TransparentIdentifier.Length
+                                                  + 1 // number suffix
```

Comment:
Actually, in some cases, you maybe have more than a single number here.
`<>h__TransparentIdentifier33` for example